pipeline-model-api/src/main/java/org/jenkinsci/plugins/pipeline/modeldefinition/ast/ModelASTWhen.java

```diff
@@ -37,60 +37,50 @@
  */
 public class ModelASTWhen extends ModelASTElement {
 
-    private List<ModelASTWhenContent> conditions = new ArrayList<>();
+    private ModelASTWhenContent condition;
 
     public ModelASTWhen(Object sourceLocation) {
         super(sourceLocation);
     }
 
-    public List<ModelASTWhenContent> getConditions() {
-        return conditions;
+    public ModelASTWhenContent getCondition() {
+        return condition;
     }
 
-    public void setConditions(List<ModelASTWhenContent> conditions) {
-        this.conditions = conditions;
+    public void setCondition(ModelASTWhenContent condition) {
+        this.condition = condition;
     }
 
     @Override
     public Object toJSON() {
-        final JSONArray a = new JSONArray();
-        for (ModelASTWhenContent c : conditions) {
-            a.add(c.toJSON());
-        }
-        return new JSONObject().accumulate("conditions", a);
+        return new JSONObject().accumulate("condition", condition.toJSON());
```

Comment:
The editor doesn't yet have `when` support in the first place, I believe - @kzantow, can you verify?